File: Software University/04) PHP Web Development Basics/16) OOP ADVANCED - Exercises/07 Food Shortage/Models/Person.php
<?php


namespace Models;


abstract class Person implements IBuyer
{
    protected $name;
    protected $age;

    public function __construct(string $name, int $age)
    {
        $this->setName($name)
            ->setAge($age);
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function setName($name): Person
    {
        $this->name = $name;
        return $this;
    }

    public function getAge(): int
    {
        return $this->age;
    }

    public function setAge($age): Person
    {
        $this->age = $age;
        return $this;
    }
}

File: Software_University/10_Java_OOP_February_2019/09. Inheritance - Exercises/01. Person/people/Person.java
package people;

public class Person {

    private static final int MIN_NAME_LEN = 3;

    private String name;

    private int age;

    public Person(String name, int age) throws IllegalArgumentException {
        this.setName(name);
        this.setAge(age);
    }

    @Override
    public String toString() {
        return String.format(
                "Name: %s, Age: %d",
                this.getName(),
                this.getAge());
    }

    public String getName() {
        return name;
    }

    public void setName(String name) throws IllegalArgumentException {
        if (name.length() < MIN_NAME_LEN) {
            throw new IllegalArgumentException("Name's length should not be less than 3 symbols!");
        }
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) throws IllegalArgumentException {
        if (age < 1) {
            throw new IllegalArgumentException("Age must be positive!");
        }
        this.age = age;
    }
}
